tests/unit/Core/Framework/Sso/Config/LoginConfigServiceTest.php

```diff
@@ -387,47 +343,61 @@ public function testCreateRedirectUrl(string $random, LoginConfig $loginConfig,
+                'rawConfig' => [
+                    'use_default' => true,
+                    'client_id' => 'justAClientID',
+                    'client_secret' => 'justAClientSecret',
+                    'redirect_uri' => 'http://justARedirectUri.org',
+                    'base_url' => 'http://justABaseUrl.net',
+                    'authorize_path' => '/authorize',
+                    'token_path' => '/token',
+                    'jwks_path' => '/jwks.json',
+                    'scope' => 'scope',
+                    'register_url' => 'http://register.url',
+                ],
+                'expectedUrl' => 'http://justABaseUrl.net/authorize?client_id=justAClientID&redirect_uri=http%3A%2F%2FjustARedirectUri.org&response_type=code&scope=scope&state=api.oauth.sso.code%3Frdm%3DjustARandomString',
             ],
 
             'Test case two' => [
                 'random' => 'justARandomString',
-                'loginConfig' => new LoginConfig(
-                    true,
-                    'anotherClientID',
-                    'anotherClientSecret',
-                    'http://another-redirect-url.org',
-                    'http://another-base-url.net',
-                    '/authorize',
-                    '/token',
-                    '/jwks.json',
-                    'scope',
-                    'http://register.url',
-                ),
-                'expectedUrl' => 'http://another-base-url.net/authorize?client_id=anotherClientID&redirect_uri=http%3A%2F%2Fanother-redirect-url.org&response_type=code&scope=scope&state=http%3A%2F%2Fapp.url%2Fapi%2Foauth%2Fsso%2Fcode%3Frdm%3DjustARandomString',
+                'rawConfig' => [
+                    'use_default' => true,
+                    'client_id' => 'anotherClientID',
+                    'client_secret' => 'anotherClientSecret',
+                    'redirect_uri' => 'http://another-redirect-url.org',
+                    'base_url' => 'http://another-base-url.net',
+                    'authorize_path' => '/authorize',
+                    'token_path' => '/token',
+                    'jwks_path' => '/jwks.json',
+                    'scope' => 'scope',
+                    'register_url' => 'http://register.url',
+                ],
+                'expectedUrl' => 'http://another-base-url.net/authorize?client_id=anotherClientID&redirect_uri=http%3A%2F%2Fanother-redirect-url.org&response_type=code&scope=scope&state=api.oauth.sso.code%3Frdm%3DjustARandomString',
             ],
         ];
     }
 
+    /**
+     * @param array<string, string|bool> $rawConfig
+     */
+    public function createLoginConfigService(array $rawConfig): LoginConfigService
+    {
+        $router = $this->createMock(RouterInterface::class);
+        $router->method('generate')->willReturnCallback(function ($name, $parameter) {
+            return $name . '?' . \http_build_query($parameter);
+        });
+
+        // @phpstan-ignore-next-line argument.type
```

Comment:
would you mind changing it to this? 
```suggestion
        // @phpstan-ignore specific.identifier.change.me (comment why we need to ignore it)
```